osu.Game.Rulesets.Taiko/Difficulty/Evaluators/ColourEvaluator.cs

```diff
@@ -0,0 +1,75 @@
+// Copyright (c) ppy Pty Ltd <contact@ppy.sh>. Licensed under the MIT Licence.
+// See the LICENCE file in the repository root for full licence text.
+
+using System;
+using osu.Game.Rulesets.Taiko.Difficulty.Preprocessing;
+using osu.Game.Rulesets.Taiko.Difficulty.Preprocessing.Colour;
+using osu.Game.Rulesets.Taiko.Difficulty.Preprocessing.Colour.Data;
+
+namespace osu.Game.Rulesets.Taiko.Difficulty.Evaluators
+{
+    public class ColourEvaluator
+    {
+        private static double sigmoid(double val, double center, double width)
+        {
+            return Math.Tanh(Math.E * -(val - center) / width);
+        }
+
+        private static double sigmoid(double val, double center, double width, double middle, double height)
+        {
+            return sigmoid(val, center, width) * (height / 2) + middle;
+        }
+
+        /// <summary>
+        /// Evaluate the difficulty of the first note of a <see cref="MonoEncoding"/>.
+        /// <param name="encoding">The encoding to evaluate.</param>
+        /// <param name="i">The index of the mono encoding within it's parent <see cref="ColourEncoding"/>.</param>
+        /// </summary>
+        public static double EvaluateDifficultyOf(MonoEncoding encoding, int i)
+        {
+            return sigmoid(i, 2, 2, 0.5, 1);
+        }
+
+        /// <summary>
+        /// Evaluate the difficulty of the first note of a <see cref="ColourEncoding"/>.
+        /// </summary>
+        /// <param name="encoding">The encoding to evaluate.</param>
+        /// <param name="i">The index of the colour encoding within it's parent <see cref="CoupledColourEncoding"/>.</param>
+        public static double EvaluateDifficultyOf(ColourEncoding encoding, int i)
+        {
+            return sigmoid(i, 2, 2, 0.5, 1);
+        }
```

Comment:
Neither of these take an encoding, so they can be removed from the method and xmldocs. Not to mention that they appear to be the exact same, so one can be removed.

Reply:
Oops missed this, earlier version uses the run length of each encoding, but it ends up being not required. They also had different parameters to sigmoid at some point, but now that both of these are the same it does make some sense to just merge them.

Reply:
With stateless colour evaluator, these now use the respective encoding types